### DISCUSSION TITLE
(feature request) select word on double click and drag to extend selection

### DISCUSSION THREAD

**Discussion by aless3** [tmux/tmux#3894] [General]
(Posting as a discussion because I read this is not yet supported, if this should be supported I will open an issue and post relevant info) 

Double click to select word works, but if I drag the mouse to extend the selection the selection does not extend. 

If I have 2 tmux panes side by side and I want to copy some text I either click and drag (and I have to be precise with the mouse) or I hold shift to "bypass tmux", but this has the cumbersome effect of selecting text from lateral panes if the text I need to copy is multi-line. 

Would such a feature be possible to add? And actually a "triple click select whole line" would be helpful but not as much as "double click word and drag". 

Thank you very much for your work!

**Comment by aless3** [comment]
well, in the end I wrote a small script installable with tmux plugin manager:
https://github.com/aless3/tmux-click-copy
I used it for a bit and except tweaking the timeout passed to sleep to one's preferences it seems stable and usable, at least I use it (obv issues and prs are welcome) 

I have not yet sent a pr to tpm or other repos to add it to any list, I don't have enough time tbh

if the admins think this discussion should be closed then close it (I'm not even sure github discussions can be closed, but whatever)

(I hope this is not considered spam because it's my plugin, in case feel free to remove the comment etc)

**Comment by orchistro** [comment]
Double click and drag to make selections word-by-word. This is what I've always wanted too!
I should try your plugin @aless3 Thanks!

**Reply by aless3** [reply]
I'm glad you find it interesting!
feel free to open issues and pr for anything!
I hope you'll find it useful!

**Comment by lourenci** [comment]
[Someone raised this issue a while ago](https://github.com/tmux/tmux/issues/682), but unfortunately, no one stepped up yet. It’s the fundamental basis of any OS regarding mouse selection, and I don’t understand why everyone is not in need of this. 🤷‍♂️ Anyway, using your plugin in meanwhile, @aless3. 🙏
